### PR TITLE
op-node: fix p2p deserialization for ecotone

### PR DESCRIPTION
**Pull request by tynes**
**Description**

Seems like ecotone updates are not caught in the p2p req/res
sync codepath? Might be a few other places that need updates
too.

### CODE REVIEW

File: op-node/p2p/sync.go [hunk 1]
```diff
@@ -655,7 +655,9 @@ func (s *SyncClient) doRequest(ctx context.Context, id peer.ID, expectedBlockNum
 
 func (s *SyncClient) readExecutionPayload(data []byte, expectedTime uint64) (*eth.ExecutionPayloadEnvelope, error) {
 	blockVersion := eth.BlockV1
-	if s.cfg.IsCanyon(expectedTime) {
+	if s.cfg.IsEcotone(expectedTime) {
+		blockVersion = eth.BlockV3
+	} else if s.cfg.IsCanyon(expectedTime) {
 		blockVersion = eth.BlockV2
 	}
```
Comment (sebastianst) [inline op-node/p2p/sync.go:662]: Should we encapsulate this into a function `rollup.Config.BlockVersion(timestamp uint64) eth.BlockVersion` similar to [`Config.NewPayloadVersion` and the other `...Version` functions](https://github.com/ethereum-optimism/optimism/pull/9253/files#diff-b6f49e11d389be9577cf5848ed47e2b1260d7ced42b1859dc8837ffb084f396a)?

Comment (tynes) [reply op-node/p2p/sync.go:662]: This is a great idea, will help to prevent us forgetting to update particular codepaths

Comment (danyalprout) [reply op-node/p2p/sync.go:662]: > Should we encapsulate this into a function rollup.Config.BlockVersion(timestamp uint64) eth.BlockVersion

I think this is a great idea. Will definitely help going forward.

For this case, Ecotone blocks should never reach this particular function. If they do -- we should likely return an error vs. attempting to decode as v3. The context is there are two different response versions [code](https://github.com/ethereum-optimism/optimism/blob/develop/op-node/p2p/sync.go#L628-L637) ([specs](https://specs.optimism.io/protocol/rollup-node-p2p.html?highlight=p2p#payload_by_number)):

* Version 0: ExecutionPayloads are returned (timestamp is used to infer v1 or v2 based on Canyon)
* Version 1: ExecutionPayloadEnvelopes are returned (currently only Ecotone)

This is the code to handle the Version 0 decoding only. It's more confusing than it should be as in hindsight I made a design mistake for Canyon by not using a different response version and inferring it from the expected timestamp of the blocks.

Comment (tynes) [reply op-node/p2p/sync.go:662]: Ok thank you for the explanation - some godocs on these functions would be super helpful to document this behavior

**Pull request closed without merging**